### ISSUE TITLE
Remote storage credentials should not be stored in plain-text

### ISSUE THREAD

**Issue by xieve** [kopia/kopia#4115]
# Reproduction steps
1. Install Kopia GUI on Windows 10
2. Connect to WebDAV repository
3. Run `kopia.exe repository status` to find config file
4. Open config file
5. Observe that `storage.config.password` contains the password to the WebDAV server in plain text

# Proposed resolution
The documentation states that the repository encryption password is already stored in the OS' credential manager, so that seems like a reasonable solution.
Alternatively, credentials could be encrypted with the same repository encryption password.

At the very least, this fact should be clearly documented and warned about when using the aforementioned GUI.

# Personal opinion
I think this is a very high severity issue. I have currently set up Kopia to back up to a Nextcloud instance that is also being used for other purposes by multiple people. Not knowing that my credentials would be stored in plain-text, I just entered my main login for WebDAV. If a malicious actor got these, they could read quite a lot of personal info, and delete or encrypt a lot of data. 

Even in the case that I had set up a new account just for Kopia, they could still encrypt or delete the repository.

**Comment by siment** [comment]
I agree, this is a very odd design choice. Especially considering the security-centric posture communicated by the project. I hope this issue can get some attention from the developers. Even (most) sFTP clients and other connection managers have long since started encrypting the credentials of bookmarked connections. 

I guess a counter argument would be that all the files in the repository are encrypted, but in any case the attackers will get write permissions to the storage and can do some real damage.

**Comment by akash-rp** [comment]
Any update on this?

**Comment by hobopanda** [comment]
Any updates on this. crazy that passwords in clear text existed in the first place.
